### ISSUE TITLE
[$500] After splitting money in workspace chat, workspace chat disappears in LHN

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#30990]
~Held on https://github.com/Expensify/Web-Expensify/pull/39529~

If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.3.96-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @aimane-chnaif 
**Slack conversation:** N/A - [reported here](https://github.com/Expensify/App/issues/30990#issuecomment-1814306511)

## Action Performed:

1. Go to https://staging.new.expensify.com/
2. Tap on a workspace chat with 2 members
3. Tap plus icon near compose
4. Tap 'Split bill or 'Request money'
5. Enter an amount
6. Tap split
7. Observe the LHN

## Expected Result:

When we create a split bill in a workspace chat with 2members, in LHN workspace chat must be displayed

## Actual Result:

When we create a split bill in a workspace chat with 2members, in LHN workspace chat disappears. It's visible after refreshing

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [x] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/96077027/f3b08810-c802-4a68-b553-0659f0d2e731

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d2443df988bfc52a</li>
        <li>Upwork Job ID: 1721868493830160384</li>
        <li>Last Price Increase: 2023-12-05</li>
    </ul>
</details>

**Comment by saikoaizen** [comment]
@izarutskaya Hello, I'm from Upwork. Can I know if the reward is still active on this issue? I'm unsure because the title has changed

**Comment by saikoaizen** [comment]
> @saikoaizen yes, open for proposals. `Help Wanted` label added as you see

Thank you for the clarification :)

**Comment by barttom** [comment]
Hi! I'm Bartek from Callstack and would like to work on this issue. I'm gonna debug this first  and will be back with the proposal ASAP

**Comment by barttom** [comment]
@izarutskaya Hey!
I'm not sure if I understand correctly what is the issue here, so please let me rephrase it:
after splitting g the bill with two other participants from chat, we should have created two individual chats for each of them,
Is that correct?

Take a look at my recording - I have two chats after splitting the bill. Is that an expected behavior?

https://github.com/Expensify/App/assets/7682108/ad040eb6-fdf1-4911-b1ff-109ea7dff0f4

**Comment by barttom** [comment]
@izarutskaya friendly bump :)

**Comment by conorpendergrast** [comment]
I've tried to reproduce and I agree with you @barttom; I'm seeing that this is working correctly and creating the request DMs correctly. I'm going to close this as not reproducible for now, but if you have more specific steps to reproduce @izarutskaya, please reply here!

**Comment by conorpendergrast** [comment]
@aimane-chnaif that's splitting in a workspace rather than a group DM. Have you also tried in a group DM, and were able to reproduce?

**Comment by barttom** [comment]
thanks, I'll try to reproduce that on workspace chat.

**Comment by barttom** [comment]
@conorpendergrast 
I'm able to reproduce issue that @aimane-chnaif mention.
It looks like something different than this mentioned in [comment](https://github.com/Expensify/App/issues/30990#issue-1981244843) that I can't reproduce, @aimane-chnaif also can't.

Can we change the description or create another issue to avoid confusion? 
I'm going to debug that after we request money from workspace chat  it disappears from LHN

**Comment by barttom** [comment]
@conorpendergrast  It's more like 'After splitting money in workspace chat, workspace chat disappears in LHN' than the current title

**Action Performed:**
1. Go to https://staging.new.expensify.com/
2. Tap on a workspace chat with 2 members
3. Tap plus icon near compose
4. Tap split bill
5. Enter an amount
6. Tap split
7. Observe the LHN

Expected Result:
When we create split bill in a workspace chat with 2members, in LHN workspace chat must be displayed

Actual Result:
When we create split bill in a workspace chat with 2members, in LHN workspace chat disappears. It's visible after refreshing

@aimane-chnaif looks good?

**Comment by barttom** [comment]
updated my last comment in point 4.
For now I found that something happens in `SidebarLinkData` component and `SidebarUtils.getOrderedReportIDs` method that creates this issue, I keep digging :)

**Comment by conorpendergrast** [comment]
Fantastic, thanks - I've updated the OP now

**Comment by barttom** [comment]
Hey!
I have some insights and question about this issue.

1. When we request money or split bill, API call RequestMoney is called.
2. Then `chatReports` from Onyx (https://github.com/Expensify/App/blob/main/src/pages/home/sidebar/SidebarLinksData.js#L199) return a different shape for the current report, e.g. `participantAccountIDs` array is empty.
3. LIst for display ius generated [here](https://github.com/Expensify/App/blob/main/src/pages/home/sidebar/SidebarLinksData.js#L92) & [here](https://github.com/Expensify/App/blob/main/src/pages/home/sidebar/SidebarLinksData.js#L73)
4. Dropped report wasn't be returned, because of a lack of data in this condition https://github.com/Expensify/App/blob/main/src/libs/ReportUtils.js#L3383

I can't find any mutation on the app side that can change in state of `chatReports` object.
@conorpendergrast Can we add someone from the internal team to check if is not a backend issue?

https://github.com/Expensify/App/assets/7682108/74eaff1f-0ef3-468d-baea-dcde3a533bf8

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by conorpendergrast** [comment]
@cristipaval want to move it internal instead? Or do you want to do anything else to test and confirm first?

**Comment by cristipaval** [comment]
Haven't got time to look into this one yet

**Comment by conorpendergrast** [comment]
Still held on https://github.com/Expensify/Web-Expensify/pull/39529

**Comment by cristipaval** [comment]
Hold removed on this one. Could we retest?

**Comment by conorpendergrast** [comment]
Re-tested, and this is working correctly! 

The workspace chat does not disappear from the LHN after the request or bill split 👍 

<img width="1014" alt="image" src="https://github.com/Expensify/App/assets/5802477/8e590d6e-bd34-4a5a-9bb3-a05d5b777d42">

(not surprising, because yes this is a dupe of https://github.com/Expensify/App/issues/30800!)

Closing out now
